src/System.Private.CoreLib/shared/System/IO/TextWriter.cs

```diff
@@ -774,6 +787,14 @@ protected override void Dispose(bool disposing)
                     ((IDisposable)_out).Dispose();
             }
 
+            // [MethodImpl(MethodImplOptions.Synchronized)]
+            public override ValueTask DisposeAsync()
+            {
+                // TODO: https://github.com/dotnet/coreclr/issues/20499
+                // Manual synchronization should be replaced by Synchronized for consistency.
+                lock (this) return _out.DisposeAsync();
```

Comment:
Same concern here - this doesn't look like it's thread safe.

Reply:
It depends on your definition of thread-safe... it's just as safe as the rest of the async methods here, which all protect the synchronous invocation under the lock and release the lock regardless of whether the async processing has completed.